pdns/nod.cc

```diff
@@ -137,8 +152,20 @@ bool PersistentSBF::snapshotCurrent(std::thread::id tid)
         // Now write it out to the file
         ofile << iss.str();
 
-        if (ofile.fail())
-          throw std::runtime_error("Failed to write to file:" + f.string());
+        if (ofile.fail()) {
+          ofile.close();
+          remove(ftmp);
```

Comment:
Shouldn't this be `unlink`? i have trouble even figuring what we're calling here.

Reply:
+1 to that, thanks